ci/docker_push.sh

```diff
@@ -15,28 +15,16 @@ do
 done
 if [ -z "$CIRCLE_PULL_REQUEST" ] && [ -z "$CIRCLE_TAG" ] && [ "$want_push" == "true" ]
 then
-   # TODO(mattklein123): Currently we are doing this push in the context of the release job which
```

Comment:
I was looking at this again, and honestly if you are already in here I think we can just fix the point release thing now if you are up for it. I would rather just do it now because any time we change these scripts something breaks, so might as well break and fix it all at the same time?

I think you have the jobs setup correct, which is that we want to run the release job both on normal branches and for tags starting with `v.`. 

I think the only changes we need are the following:
1) In this script, you would make the if statement run if branch == master or `$CIRCLE_TAG` is not empty. At the same time can you cleanup the `want_push` stuff? I don't think it's needed and can just be a comparison for branch == master. 

2) Remove this if check and TODO: https://github.com/envoyproxy/envoy/blob/master/ci/do_ci.sh#L67. This will make us do a build on tag events also.

For normal master releases, we may end up building and pushing the same SHA to docker hub. Do we need any kind of force or detection logic there? (For tag releases the build SHA will be unique)

Reply:
Made these changes.

> For normal master releases, we may end up building and pushing the same SHA to docker hub. Do we need any kind of force or detection logic there? (For tag releases the build SHA will be unique)

Not afaik. Docker tags are just mutable pointers. It would be a bit of a waste as you'd probably end up pushing mostly new image layers for the same image tag but that's probably ok for now? Can do something smarter, of course.